app/poi_api/poi_service.py

```diff
@@ -0,0 +1,106 @@
+from app import firestore_db
+from flask import jsonify
+from .poi_api import POIApi
+from .poi_suggestion import POI_suggestions
+from .poi import POI
+from .poi_errors import POINotFoundError, InvalidPOISuggestionError
+from datetime import datetime
+
+
+class POI_Service(POIApi):
+    def get_all_POI(self):
+        try:
+            poi_ref = firestore_db.collection("POI")
+            all_poi = [doc.to_dict() for doc in poi_ref.stream()]
+            list_all_poi = []
+            for poi in all_poi:
+                poi_object = self._create_poi_object(poi)
+                list_all_poi.append(poi_object)
+            return list_all_poi

```

Comment:
I know this isn't final, there's more logic to add later but in theory your could return the `poi_ref.stream()` mapping over with a parse function to instead return a stream of POI objects.